t/00unit/lib/common/url.c

```diff
@@ -29,75 +29,170 @@ static void test_normalize_path(void)
     h2o_mem_init_pool(&pool);
 
     size_t q;
-    h2o_iovec_t b = h2o_url_normalize_path(&pool, H2O_STRLIT("/"), &q);
+    size_t *norm_indexes = NULL;
+    h2o_iovec_t input;
+    h2o_iovec_t b;
+
+    input = h2o_iovec_init(H2O_STRLIT("/"));
+    b = h2o_url_normalize_path(&pool, input.base, input.len, &q, &norm_indexes);
+    ok(b.len == 1);
+    ok(memcmp(b.base, H2O_STRLIT("/")) == 0);
+    ok(q == SIZE_MAX);
+    ok(norm_indexes == NULL);
+
+    input = h2o_iovec_init(H2O_STRLIT("/."));
+    b = h2o_url_normalize_path(&pool, input.base, input.len, &q, &norm_indexes);
+    ok(b.len == 1);
+    ok(memcmp(b.base, H2O_STRLIT("/")) == 0);
+    ok(q == SIZE_MAX);
+    memset(norm_indexes, 0, input.len * sizeof(norm_indexes[0]));
```

Comment:
Also, please test the values of `norm_indexes[0]` (I believe it contains a valid value).